### PR TITLE
Json parsing

### PR DESCRIPTION
**Pull request by apan145**
Implementation of parsing of JSON files using the GSON library.
Restructuring of how databases are handled.

### CODE REVIEW

File: src/main/resources/Quests.json [hunk 1]
```diff
@@ -1,20 +1,17 @@
-{
-  "Quest": {
```
Comment (Little0smit) [inline src/main/resources/Quests.json:2]: Prefer these headings left in, as when the file is full each object will be the name of the quest/mob/shop etc

File: src/main/resources/Chunks.json [hunk 2]
```diff
@@ -1,15 +1,29 @@
...
-}
+[
+  {
+    "chunkNumber": 1,
+    "chunkName": "1",
+    "mobs": [
+      {
+        "name": "",
+        "combatLvl": 2,
+        "drops":[""]
```
Comment (Little0smit) [inline src/main/resources/Chunks.json]: No need to store drops in json file, just adds more work/duplication. Simply lookup the mobs database to get drops.

File: src/main/resources/Chunks.json [hunk 3]
```diff
@@ -1,15 +1,29 @@
...
+      ""
+    ],
+    "processingTools": [
+      ""
+    ],
+    "shops": [
+      {
+        "name": "Name",
+        "chunkLocation": [2000],
+        "stock":[""],
```
Comment (Little0smit) [inline src/main/resources/Chunks.json]: Again, if we have a db of shops, no need to add stock to 2 json files.

File: src/main/resources/Mobs.json [hunk 4]
```diff
@@ -1,7 +1,22 @@
-{
-  "Mob":{
```
Comment (Little0smit) [inline src/main/resources/Mobs.json:2]: Keep object name, can be replaced with mobID or name to make file easier to navigate

File: src/main/resources/Mobs.json [hunk 5]
```diff
@@ -1,7 +1,22 @@
...
+    "combatLvl": 33,
+    "drops":["flour", "bucket"]
+  },
+  {
+    "name": "chicken",
+    "combatLvl": 99,
+    "drops":["feather"]
+  },
+  {
+    "name": "yew tree",
```
Comment (Little0smit) [inline src/main/resources/Mobs.json]: Why is a yew tree a mob?

File: src/main/resources/Processes.json [hunk 6]
```diff
@@ -1,9 +1,11 @@
-{
-  "Processing": {
```
Comment (Little0smit) [inline src/main/resources/Processes.json:2]: Keep obj name.

File: src/main/resources/Shops.json [hunk 7]
```diff
@@ -1,7 +1,8 @@
-{
-  "Shop": {
```
Comment (Little0smit) [inline src/main/resources/Shops.json:2]: Keep headings

**Review by Little0smit — CHANGES_REQUESTED** [review]
Quick check on mobile but a few changes noticed. Will have a proper inspection on PC tomorrow (gmt)

File: src/main/resources/Quests.json (hunk 1 above)
Comment (apan145) [reply src/main/resources/Quests.json:2]: You can't have headings for objects in arrays.

File: src/main/resources/Mobs.json (hunk 5 above)
Comment (apan145) [reply src/main/resources/Mobs.json]: I was testing some stuff and forgot to delete it

File: src/main/resources/Processes.json (hunk 6 above)
Comment (apan145) [reply src/main/resources/Processes.json:2]: Same as above

File: src/main/resources/Shops.json (hunk 7 above)
Comment (apan145) [reply src/main/resources/Shops.json:2]: Same as above

File: src/main/resources/Quests.json (hunk 1 above)
Comment (Little0smit) [reply src/main/resources/Quests.json:2]: Okay, was originally just gonna store all quest objects within an object, but guess an array doesn't hurt it much. 
There any large advantage of using an array that we gain from losing object names of each object?

File: src/main/resources/Chunks.json (hunk 3 above)
Comment (apan145) [reply src/main/resources/Chunks.json]: Solved

File: src/main/resources/Chunks.json (hunk 2 above)
Comment (apan145) [reply src/main/resources/Chunks.json]: Solved

File: src/main/resources/Mobs.json (hunk 4 above)
Comment (apan145) [reply src/main/resources/Mobs.json:2]: Same as above

File: src/main/resources/Quests.json (hunk 1 above)
Comment (apan145) [reply src/main/resources/Quests.json:2]: It'd simplify the data structure. When the JSON parsing library parses the file, it tries to store the different fields of the object in variables with the same name, eg if there is a field called questName, it'll store the fields value in the variable questName if the output class has a variable named that. If we keep the headers, the class that we'd parse to would have to have a different variable for each quest, shop, chunk ect.

Comment (Little0smit) [reply src/main/resources/Quests.json:2]: Ok fair enough

**Review by Little0smit — APPROVED** [review]

**Pull request merged**